orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/CreateServerGroupTaskSpec.groovy

```diff
@@ -76,4 +79,577 @@ class CreateServerGroupTaskSpec extends Specification {
+    bakeSynthetic1.id = UUID.randomUUID()
+    bakeSynthetic1.parentStageId = bakeStage1.id
+    bakeStage1.execution.stages << bakeSynthetic1
+
+    def childTrigger = [
+      parentExecution: parentPipeline
+    ]
+    def childPipeline = new Pipeline.Builder().withName("child").withTrigger(childTrigger).build()
+    def deployStage = new PipelineStage(childPipeline, "createServerGroup", deployConfig)
+    deployStage.execution.stages.add(deployStage)
+
+    def manualJudgmentStage = new PipelineStage(deployStage.execution, "manualJudgment")
+    manualJudgmentStage.id = UUID.randomUUID()
+    manualJudgmentStage.refId = "1"
+    deployStage.execution.stages << manualJudgmentStage
+    deployStage.requisiteStageRefIds = [manualJudgmentStage.refId]
+
+    def deployTask = new CreateServerGroupTask(kato: katoService, serverGroupCreators: [new AmazonServerGroupCreator(mortService: mortService), new GoogleServerGroupCreator()])
+
+    when:
+    def result = deployTask.execute(deployStage)
+
+    then:
+    _ * mortService.getAccountDetails("abc") >> [:]
+    1 * katoService.requestOperations(operationCloudProvider, {
+      return it[createOpIndex]?.createServerGroup?.get(imageAttributeKey) == expectedImageId
+    }) >> { Observable.from(taskId) }
+    // This helps avoid an NPE within CreateServerGroupTask; this results in better error-reporting on a test failure.
+    _ * katoService.requestOperations(operationCloudProvider, _) >> { Observable.from(taskId) }
+    result?.stageOutputs == baseOutput + ["kato.result.expected": katoResultExpected]
+
+    where:
+    operationCloudProvider | bakeCloudProvider | createOpIndex | imageAttributeKey | katoResultExpected || expectedImageId
+    "gce"                  | "gce"             | 0             | "image"           | false              || "parent-name"
+    "gce"                  | null              | 0             | "image"           | false              || "parent-name"
+    "aws"                  | "aws"             | 1             | "imageId"         | true               || "parent-name"
+    "aws"                  | null              | 1             | "imageId"         | true               || "parent-name"
+  }
+
+  @Unroll
+  def "image for #bakeCloudProvider in grandparent pipeline stage context is resolved by create stage for #operationCloudProvider"() {
+    /*
+      grandparentPipeline: bake
+                           | trigger
+                           v
+                             parentPipeline:
+                                             | trigger
+                                             v
+                                               childPipeline: manualJudgment -> createServerGroup
+     */
+    given:
+    KatoService katoService = Mock(KatoService)
+    MortService mortService = Mock(MortService)
+    def deployRegion = "us-east-1"
+    def deployConfig = [
```

Comment:
Would be nice to eliminate some of the repeated config in these test cases ... 

Probably beneficial to have some helpers that build these graphs and stitch stages together by refId/requisiteIds.